lib/rules/no-magic-number.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * @fileoverview Rule to flag statements that use magic numbers (adapted from https://github.com/danielstjules/buddy.js)
+ * @author Vincent Lemeunier
+ * See LICENSE file in root directory for full license.
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = function(context) {
+    var config = context.options[0] || {},
+        ignore = config.ignore || [0, 1, 2],
+        detectObjects = !!config.detectObjects,
+        enforceConst = !!config.enforceConst;
+
+    /**
+     * Returns whether the node is number literal
+     * @param {Node} node - the node literal being evaluated
+     * @returns {boolean} true if the node is a number literal
+     */
+    function isNumber(node) {
+        return typeof node.value === "number";
+    }
+
+    /**
+     * Returns whether the number should be ignored
+     * @param {number} num - the number
+     * @returns {boolean} true if the number should be ignored
+     */
+    function shouldIgnoreNumber(num) {
+        return ignore.indexOf(num) !== -1;
+    }
+
+
+    return {
+        "Literal": function(node) {
+            var parent = node.parent,
+                okTypes = detectObjects ? [] : ["ObjectExpression", "Property", "AssignmentExpression"];
+
+            if (!isNumber(node)) {
+                return;
+            } else if (shouldIgnoreNumber(node.value)) {
+                return;
+            }
```

Comment:
You can combine this into one condition: `if (!isNumber(node) || shouldIgnoreNumber(node.value)) {`. That's how I'd do it personally; feel free to ignore.